azurerm/internal/services/machinelearning/machine_learning_inference_cluster_resource.go

```diff
@@ -0,0 +1,502 @@
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"machine_learning_workspace_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"kubernetes_cluster_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.KubernetesClusterID,
+			},
+
+			"cluster_purpose": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "Dev",
+			},
+
+			"node_pool_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NodePoolID,
+			},
+
+			"identity": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(machinelearningservices.SystemAssigned),
+							}, false),
+						},
+						"principal_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"ssl": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cert": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "",
+						},
+						"key": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "",
+						},
+						"cname": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "",
+						},
+					},
+				},
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceAksInferenceClusterCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	mlWorkspacesClient := meta.(*clients.Client).MachineLearning.WorkspacesClient
+	mlComputeClient := meta.(*clients.Client).MachineLearning.MachineLearningComputeClient
+	aksClient := meta.(*clients.Client).Containers.KubernetesClustersClient
+	poolsClient := meta.(*clients.Client).Containers.AgentPoolsClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	// Define Inference Cluster Name
+	name := d.Get("name").(string)
+
+	// Get Machine Learning Workspace Name and Resource Group from ID
+	ml_workspace_id := d.Get("machine_learning_workspace_id").(string)
+	ws_id, err := parse.WorkspaceID(ml_workspace_id)
+	if err != nil {
+		return err
+	}
+
+	workspace_name := ws_id.Name
+	resource_group_name := ws_id.ResourceGroup
```

Comment:
Variables should be camelCase, I'd use workspaceID directly instead of initiating the variables here already
```suggestion
	workspaceIDUnparsed := d.Get("machine_learning_workspace_id").(string)
	workspaceID, err := parse.WorkspaceID(workspaceIDUnparsed)
	if err != nil {
		return err
	}
```